### ISSUE TITLE
are we sure left padding is useful here?

### ISSUE THREAD

**Issue by Jemoka** [alliegriffith/perez_reimplementation#1]
...in particular, left padding makes rollouts work, but may screw up probability calculations if attention mask is not propagated correctly.

https://github.com/alliegriffith/perez_reimplementation/blob/bb6201d149409e102ef43b27e258a76d1a11eab9/perez.py#L61-L62

**Comment by alliegriffith** [comment]
I am keeping the left padding for now but tried to make sure the attention mask is being propagated/handled properly when calculating adv/def logits for KL divergence, when calculating critic value estimates, and when calculating the log probs of the adversarial utterances.

**Comment by Jemoka** [comment]
epic; sounds good. I'm not too worried about this
